android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/MedeaFlutterWebrtcPlugin.kt

```diff
@@ -9,26 +11,51 @@ import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
 import io.flutter.plugin.common.BinaryMessenger
 import io.flutter.view.TextureRegistry
 
+private val TAG = MedeaFlutterWebrtcPlugin::class.java.simpleName
+
 class MedeaFlutterWebrtcPlugin : FlutterPlugin, ActivityAware {
   private var peerConnectionFactory: PeerConnectionFactoryController? = null
   private var mediaDevices: MediaDevicesController? = null
   private var videoRendererFactory: VideoRendererFactoryController? = null
   private var messenger: BinaryMessenger? = null
   private var state: State? = null
   private var textureRegistry: TextureRegistry? = null
+  private var permissions: Permissions? = null
+  private var activityPluginBinding: ActivityPluginBinding? = null
 
   override fun onAttachedToEngine(registrar: FlutterPlugin.FlutterPluginBinding) {
+    Log.i(TAG, "Attached to engine")
+
     messenger = registrar.binaryMessenger
     state = State(registrar.applicationContext)
     textureRegistry = registrar.textureRegistry
   }
 
-  override fun onDetachedFromEngine(registrar: FlutterPlugin.FlutterPluginBinding) {}
+  override fun onDetachedFromEngine(registrar: FlutterPlugin.FlutterPluginBinding) {
+    Log.i(TAG, "Detached from engine")
+
+    ControllerRegistry.disposeAll()
+    mediaDevices?.dispose()
+    videoRendererFactory?.dispose()
+    peerConnectionFactory?.dispose()
```

Comment:
Maybe make them `IdentifyableController`s and dispose in `ControllerRegistry.disposeAll()`? In that case we will make controllers disposing logic more consistent without unnecessary exceptions. So all controllers will implement `IdentifiableController`s and would be registered in `ControllerRegistry`. Or there is reason to not do it?